### ISSUE TITLE
float32 BJD numerical errors

### ISSUE THREAD

**Issue by lmarthur** [SamMerc/sajax#24]
sajax/planet.py casts the time axis to float32 before computing planet positions:

- compute_planet_sky_positions: jnp.asarray(times, dtype=jnp.float32) (planet.py:236)
- build_transit_model: times_jax = jnp.asarray(times, dtype=jnp.float32) (planet.py:338)
- core.py:1107-1111 hands it absolute times already cast down (times_oversampled = times_arr.astype(np.float32))

float32 has 24 mantissa bits, so at BJD 0f 2460000 the spacing between values is nearly six hours. The mean anomaly is then formed as M = (2*pi/period)(time − t_peri) (planet.py:182) — a cancelling subtraction of two numbers near 2.46e6. 

One potential fix is to reduce the epoch in float64 before anything reaches JAX by picking a static t_ref (e.g. np.floor(times.min())) and working in times − t_ref throughout. Another potential step would be to remove the  hardcoded dtype=jnp.float32 so jax_enable_x64 works for long baselines. A warning that flags when the maximum time values become large enough to affect the numerical precision could also be useful.
